### PR TITLE
Skip update input when value unchanged

### LINKED ISSUE #297: input type="number" erase bug in Chrome
### Environment

* Elixir version (elixir -v): 1.9.1
* Phoenix version (mix deps): 1.4.8
* NodeJS version (node -v): 12.7.0
* NPM version (npm -v): 6.10.0 
* Operating system: Ubuntu 19.04
* Browser: Chromium: Version 76.0.3809.87 (64-bit)

### Actual behavior

I've created a [fork of the phoenix_live_view_example repo](https://github.com/LightningK0ala/phoenix_live_view_example) to show this problem. You can clone the repo, follow the setup instructions to run it and go to [http://localhost:4000/users/new](http://localhost:4000/users/new) on a Chrome browser.

1. Type "10.00" in the "Test" input field.
2. Erase the last 2 decimal digits.

As soon as you reach the dot, the cursor will jump to the beginning and the value will change to "10".

ezgif com-video-to-gif

This issue only happens on Chrome, I've tried it both on a Mac and Ubuntu.
Firefox doesn't have this issue.
Note I don't have any browser plugins installed on Chrome.

### Expected behavior

When erasing and reaching the dot, the cursor should not jump and the value should remain "10."

### PR DESCRIPTION
**Pull request by LightningK0ala**
This PR closes #297.

There is a competing PR #299 that unfortunately didn't solve the problem for me but thanks to @snewcomer I looked further into the problem. It seems that the issue is in the way Chrome handles the changes made by morphdom. The solution I found was to fix the checks in `onBeforeElUpdated` and prevent unnecessary morphing from being done.

When a user types into an `input` element, the value is held in the element property, i.e. `el.value`. During a live "validate" round-trip, the incoming diff will have the value set as an attribute. This means that the equality check for input elements will return false.

In addition, the existing input handling logic in `onBeforeElUpdated` can set PHX_HAS_FOCUSED and PHX_HAS_SUBMITTED which will also cause the equality check to fail.

In order to do a proper check on input values, I added an additional check after the input handling logic that deep clones input elements and sets the value attribute according to the value property.

A side effect of this fix is that inputs which haven't changed are no longer unnecessarily updated by morphdom. 

Finally, I also added a bit of code to set `target.value` in `mergeAttrs` such that element value property is kept in sync with the value attribute. This enables the backend to change the value of the input through transformations, i.e. adding parenthesis / hyphens when typing a US phone number.

### CODE REVIEW

**Comment by snewcomer** [comment]
@LightningK0ala Thanks for providing more food for thought here!  I still need to dig into some of the ideas here. A couple of initial additions to the conversation. 

1.  morphdom is doing what it is told.  So not necessarily a bug in morphdom.
2. #299 solves 1 part of the issue - adding a decimal or `,`. But you are right, not the jump to the start when deleting is still and issue.  I need to look at what the full changeset looks like to solve the bug.

B/c there are no issues in Firefox/Safari, it makes me think any changes to handle this should be much smaller in scope.  Definitely not saying this PR won't be our end solution.  Just trying to add more thoughts to the conversation.  Nice work 👍

**Comment by LightningK0ala** [comment]
Thanks @snewcomer! Agreed it's not necessarily a bug in morphdom, I didn't dig into that enough. Let me know if you find something else, I'll be glad to help with more testing.

File: assets/js/phoenix_live_view.js [hunk 1]
```diff
@@ -751,6 +753,17 @@ let DOM = {
...
         DOM.discardError(toEl)
 
+        // equality check with input value
+        if (fromEl.nodeName == "INPUT") {
+          const fromCopy = fromEl.cloneNode(true)
+          // input value comes from property
+          fromCopy.setAttribute("value", fromCopy.value)
+          if (fromCopy.isEqualNode(toEl)) {
+            // avoid extra attribute change, fixes Chrome decimal erase bug
+            return false
```
Comment (snewcomer) [inline assets/js/phoenix_live_view.js]: Would this `return false` for pretty much every input change?  Would it then be better to just let morphdom do its thing on `blur`?

**Comment by LightningK0ala** [comment]
Closed, better solution: https://github.com/phoenixframework/phoenix_live_view/pull/308

**Pull request closed without merging**